app/models/certificate.server.ts

```diff
@@ -0,0 +1,64 @@
+import type { Certificate } from '@prisma/client';
+
+import { prisma } from '~/db.server';
+
+export type { Certificate } from '@prisma/client';
+
+export async function getCertificateByUsername(username: Certificate['username']) {
```

Comment:
Isn't it better to delete the certificate by `id` rather than a username, since as far as I understand that is the `PK` for it?

Reply:
My thinking is that `username` is more accessible than finding out the id/index of a certificate. We only record one (current) certificate per user, so it should be fine.

Reply:
`username` acts kind of an id, because it was set that students won't have the same `email address`, so it should work, but I do understand that ID would be a standard and good practice to do

Reply:
A database table always needs something as an unique identifier (primary key) for the row, which can often be an index number like `id`.
You can argue that we should keep `username` as the primary keep instead of using id, which we can change in a different issue, because that will require change in schema.

Reply:
> A database table always needs something as an unique identifier (primary key) for the row, which can often be an index number like `id`. You can argue that we should keep `username` as the primary key instead of using `id`, which we can change in a different issue, because that will require change in schema.

My point is that it is best to use the primary key as an id for certain entities in a table, and doing otherwise might lead to confusion among developers and feels like an antipattern to me. 

Again, I am not a big DB guy so my knowledge is limited, I am curious what @humphd @dadolhay might think on it

Reply:
I think it's best to use index as to username. Not just for helping developers out in the future, but also for having a database that expands overtime

Reply:
I don't think it is confusing to say that you want to get a certificate for a user

Reply:
It's okay to use a username if it's always going to stay unique. Since we are building for Seneca, I guess it will always stay unique.

Reply:
Yea, I think it will. Also, we can always change this if someday we want to have multiple certificates for a user

Reply:
There is a difference in speed if you use an integer (id) as to a varchar (username); it has something to do with the space they take. Not sure if our database is going to be big enough that the difference will be apparent, but I do recommend it just so we don't have to worry about it later down the line

Reply:
If you are to use the function, which way is more useful to you?

Reply:
Username makes more sense to me

Reply:
Actually now that I see it... I don't see an `id` for user, only a `username` assigned as a primary key. I'll just go with username, because I don't want to burden you with changing the schema to include an `id` just for the extra speed

Reply:
I think the point about speed is applicable to any PK(primary key). For example, ASP.NET has a function that finds an item by PK, which is faster than other methods. There is a choice to make` username `the PK in `Certificate` model.

In case there's a confusion, each table has their own PK, so what goes on in `User` model/table has no real bearing on what's the PK in other models/tables.

Reply:
I don't know if it changes the speed when reading through a string vs an int. Anyways, there shouldn't be an issue in terms of functionality

Reply:
Yeah, I meant making `username` a PK so this function potentially runs faster as it is.

Reply:
I think that should be tackled in a different issue